packages/devextreme/js/__internal/ui/overlay/m_overlay.ts

```diff
@@ -452,48 +518,61 @@ class Overlay<
     return visible ? this._show() : this._hide();
   }
 
-  _getAnimationConfig() {
-    return this._getOptionValue('animation', this);
+  _getAnimationConfig(): dxOverlayAnimation {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+    return this._getOptionValue('animation', this) ?? {};
   }
 
   // eslint-disable-next-line @typescript-eslint/no-unused-vars, class-methods-use-this
   _toggleBodyScroll(enabled?: boolean): void {}
 
   _animateShowing(): void {
-    const animation = this._getAnimationConfig() ?? {};
+    const animation = this._getAnimationConfig();
     const showAnimation = this._normalizeAnimation(animation.show, 'to');
     const startShowAnimation = showAnimation?.start ?? noop;
     const completeShowAnimation = showAnimation?.complete ?? noop;
 
+    const completeCallback = (element: HTMLElement, config: AnimationConfig): void => {
+      if (this._isAnimationPaused) {
+        return;
+      }
+
+      if (this.option('focusStateEnabled')) {
```

Comment:
this.option('optionName') loses typings

```suggestion
      const { focusStateEnabled } = this.option();
      
      if (focusStateEnabled)) {
```

Reply:
eca8905e08c0af1d0eedada4c1e167e77335af2c, 54fb3761139b89233aabdcf60cfc00701960f9b6, 331517e56476f40cbf591e671dcd6b8e43edb8ca, 38d39427a4658cbf6852f4fe72f979b6c9c54a49